packages/babel-preset-expo/src/restricted-react-api-plugin.ts

```diff
@@ -83,6 +95,23 @@ export function environmentRestrictedReactAPIsPlugin(
           });
         }
       },
+
+      // Match against `var _useState = useState(0),`
+      VariableDeclarator(path) {
+        const importedSpecifiers: undefined | Map<string, Set<string>> =
+          path.scope.getData('forbiddenImports');
+        if (!importedSpecifiers) return;
+        importedSpecifiers.forEach((forbiddenApis, importName) => {
+          if (t.isCallExpression(path.node.init) && t.isIdentifier(path.node.init.callee)) {
+            if (forbiddenApis.has(path.node.init.callee.name)) {
+              throw path.buildCodeFrameError(
+                `Client-only "useState" API cannot be used in a React server component. Add the "use client" directive to the top of this file or one of the parent files to enable running this stateful code on a user's device.`
```

Comment:
What is a "parent file"?

Reply:
A file that imports this file. The verbiage comes from next swc implementation which is cited in the comments for this plugin https://github.com/vercel/next.js/blob/d43a387d271263f2c1c4da6b9db826e382fc489c/packages/next-swc/crates/next-custom-transforms/src/transforms/react_server_components.rs#L97

Reply:
OK, I would say more clearly: "one of the files that imports this file". "Parent file" is confusing in the context of a file system where it more likely implies the parent directory.

Reply:
I still think "parent file" is a misleading term. "dependent module" or the suggestion I wrote earlier would be more accurate and unambiguous.